pinterest/gestalt
# Layer: fix issue with Flyout when zIndex gets set

When we set a zIndex on a `<Layer />` which wraps a `<Flyout />`, the flyout currently disappears.
This happens because of the `position: absolute` which is implicitly set when we set `zIndex` on `<Layer />`


## Test Plan

For one of the examples in https://gestalt.netlify.com/Flyout

1. Set `zIndex` prop on the wrapping `<Layer />`
```
<Layer zIndex={new FixedZIndex(2)}>
  <Flyout
```
2. Ensure the `Flyout` shows on the screen

### Before: Flyout no longer shows shows
![Screen Shot 2020-10-19 at 1 48 59 PM](https://user-images.githubusercontent.com/127199/96510106-e58e8a00-1211-11eb-8dc9-5c8246786597.png)

### After: Flyout shows
![Screen Shot 2020-10-19 at 1 48 53 PM](https://user-images.githubusercontent.com/127199/96510118-e7584d80-1211-11eb-9174-5d59b12b0c7f.png)
